modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -171,6 +176,21 @@ func isAllowedAddress(ctx sdk.Context, receiver string, allowedAddrs []string) b
 	return false
 }
 
+// areHopsAllowed returns whether the provided slice of Hop matches one of the allowed ones.
+func areHopsAllowed(provided []Hop, allowed []*Hops) bool {
+	if len(allowed) == 0 {
```

Comment:
shouldn't this be checking len of provided (i.e. msgTransfer hops)? 

If the length of the msg hops is zero, then it should by default be allowed. Otherwise, we must ensure that the provided hops are specified within the authz allocation.

Reply:
Two things here

* Since `allowed_forwarding_hops` is not a mandatory field, we check whether it's empty because, in that case, all hops are allowed (in my understanding

* > If the length of the msg hops is zero, then it should by default be allowed.
That's the part I'm not sure and I asked the same here: https://github.com/cosmos/ibc-go/pull/6627/files#r1645684477 
If `allowed_forwarding_hops` is not empty, but the msg contains no hops, should we accept it?

Reply:
I would like a definitive answer on this as well, but my understanding here was that if the allowed hops are set, then those hops needs to be part of the message. Meaning, you shouldn't be allowed to send through any other paths (including one without hops). To me, the allowed_forwarding_hops guarantees that when you send a token, it will end up on the expected chain with the expected denom. If that is not the case, it would be good to get that clarified :)

Reply:
Done (I didn't add the `if len(allowed) == 0`) because that's implicit (if `provided` is not empty, but `allowed` is, we directly skip the for loop and return false. 

I've also added a test case to cover it.

Thanks for the clarification!

Reply:
The only potential risk there is that would allow someone to spoof the address (e.g. a fake chain) and send it there. The only way to truly enforce that you can only send to the correct chain would then be to use both port/channel (to avoid the case above) and hops, right? 

I can see both sides of this, and neither is 100% intuitive (to me). So I think we just need to carefully document the behavior so nobody gets surprised :)